### PR TITLE
Misc clean

### PR DESCRIPTION
**Pull request by tamird**
Some stuff I found while working on errcheck

### CODE REVIEW

**Comment by spencerkimball** [comment]
LGTM

File: proto/Makefile [hunk 1]
```diff
@@ -16,17 +16,15 @@
 # Author: Spencer Kimball (spencer.kimball@gmail.com)
 # Author: Andrew Bonventre (andybons@gmail.com)
 
-PWD=$(shell pwd)
-
-PROTOC      := $(PWD)/../../../../../bin/protoc
+PROTOC      := $(GOPATH)/bin/protoc
```
Comment (petermattis) [inline proto/Makefile]: FYI, `GOPATH` might contain multiple entries which will break this.

**Comment by petermattis** [comment]
LGTM modulo the `rpc/codec` Makefile change.

File: gossip/group.go [hunk 2]
```diff
@@ -202,13 +202,12 @@ func (g *group) addInfo(i *info) (contentsChanged bool, err error) {
 	// value of prospective info is lower, take the minimum of the two
 	// Hops values.
 	if existingInfo, ok := g.Infos[i.Key]; ok {
-		if existingInfo.Timestamp < i.Timestamp ||
-			(existingInfo.Timestamp == i.Timestamp && existingInfo.Hops > i.Hops) {
-			g.removeInternal(existingInfo)
-		} else {
-			err = util.Errorf("current group info %+v newer than proposed info %+v", existingInfo, i)
+		if i.Timestamp < existingInfo.Timestamp ||
+			(i.Timestamp == existingInfo.Timestamp && i.Hops >= existingInfo.Hops) {
```
Comment (tschottdorf) [inline gossip/group.go]: `i.Timestamp == existingInfo.Timestamp` and `i.Hops == existingInfo.Hops` should trigger an update? If so, add a comment to explain why.

Comment (tschottdorf) [inline gossip/group.go]: oh wait, it switched. Disregard that.

**Comment by tschottdorf** [comment]
LGTM

File: proto/Makefile (hunk 1 above)
Comment (tamird) [inline proto/Makefile]: :crying_cat_face: 

we also have similar problems in godeps.sh - is there a reasonable way to handle this?

Comment (petermattis) [inline proto/Makefile]: In shell (err, bash) you can easily strip off everything after the first entry: `${GOPATH%%:*}`. Probably something similar can be done in make.

Comment (tamird) [inline proto/Makefile]: I've sort-of punted on this by returning the old logic that uses relative paths for GOPATH, which is what godeps.sh does. Hopefully we can use GOBIN to fix this stuff globally in the future.

**Pull request merged**